-help panicking with Go 1.7

After upgrading to Go 1.7 (and 1.7.1), `-help` gives a panic:

```
$ brew switch go 1.6.3
Cleaning /usr/local/Cellar/go/1.6.3
Cleaning /usr/local/Cellar/go/1.7
Cleaning /usr/local/Cellar/go/1.7.1
3 links created for /usr/local/Cellar/go/1.6.3
$ go build
$ ./gosonic -help
Usage of ./gosonic:
  -dbpath
    	Change value of DbPath. (default ./devDb)
  -disabledownsampling
    	Change value of DisableDownsampling.
  -disablevalidation
    	Change value of DisableValidation. (default true)
...

Generated environment variables:
   GOSONIC_PORT
   GOSONIC_MUSICFOLDER
   GOSONIC_DBPATH
...

$ brew switch go 1.7.1
Cleaning /usr/local/Cellar/go/1.6.3
Cleaning /usr/local/Cellar/go/1.7
Cleaning /usr/local/Cellar/go/1.7.1
3 links created for /usr/local/Cellar/go/1.7.1
$ go build
$ ./gosonic -help
Usage of ./gosonic:
panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 1 [running]:
panic(0x5c8280, 0xc4202c0360)
	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
reflect.valueInterface(0x0, 0x0, 0x0, 0x28744d88401, 0x0, 0xd6e84)
	/usr/local/Cellar/go/1.7.1/libexec/src/reflect/value.go:912 +0x204
reflect.Value.Interface(0x0, 0x0, 0x0, 0x11bcb, 0x6089a0)
	/usr/local/Cellar/go/1.7.1/libexec/src/reflect/value.go:907 +0x44
github.com/deluan/gosonic/vendor/github.com/fatih/structs.(*Field).Value(0xc4202d0360, 0xf547, 0x5dc700)
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/vendor/github.com/fatih/structs/field.go:31 +0x40
github.com/deluan/gosonic/vendor/github.com/koding/multiconfig.(*fieldValue).String(0xc4202d0360, 0x5ef2a0, 0xc4202d0360)
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/vendor/github.com/koding/multiconfig/flag.go:151 +0x2f
flag.isZeroValue(0xc4202ab180, 0xc420297f00, 0x7, 0xc4202c0180)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:393 +0x12d
flag.(*FlagSet).PrintDefaults.func1(0xc4202ab180)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:467 +0x1d5
flag.(*FlagSet).VisitAll(0xc4202bcea0, 0xc420053ab8)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:323 +0x67
flag.(*FlagSet).PrintDefaults(0xc4202bcea0)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:476 +0x46
github.com/deluan/gosonic/vendor/github.com/koding/multiconfig.(*FlagLoader).Load.func1()
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/vendor/github.com/koding/multiconfig/flag.go:67 +0x118
flag.(*FlagSet).usage(0xc4202bcea0)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:828 +0x5c
flag.(*FlagSet).parseOne(0xc4202bcea0, 0x617be0, 0x1, 0xc4202d40f0)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:870 +0xb4c
flag.(*FlagSet).Parse(0xc4202bcea0, 0xc420094070, 0x1, 0x1, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.1/libexec/src/flag/flag.go:915 +0x60
github.com/deluan/gosonic/vendor/github.com/koding/multiconfig.(*FlagLoader).Load(0xc4202bb900, 0x584160, 0xc420121810, 0xc420139ee0, 0x0)
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/vendor/github.com/koding/multiconfig/flag.go:82 +0x25f
github.com/deluan/gosonic/conf.LoadFromFlags()
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/conf/configuration.go:33 +0x79
main.main()
	/Users/deluan/Development/go/src/github.com/deluan/gosonic/main.go:15 +0x2b
```

For reference, my code is available here: https://github.com/deluan/gosonic/blob/master/conf/configuration.go

--- a/flag.go
+++ b/flag.go
@@ -134,36 +134,44 @@ func (f *FlagLoader) processField(fieldName string, field *structs.Field) error 
 }
 
 // fieldValue satisfies the flag.Value and flag.Getter interfaces
-type fieldValue structs.Field
+type fieldValue struct {
+	field *structs.Field
+}
 
 func newFieldValue(f *structs.Field) *fieldValue {
-	fl := fieldValue(*f)
-	return &fl
+	return &fieldValue{
+		field: f,
+	}
 }
 
 func (f *fieldValue) Set(val string) error {
-	field := (*structs.Field)(f)
-	return fieldSet(field, val)
+	return fieldSet(f.field, val)
 }
 
 func (f *fieldValue) String() string {
-	fl := (*structs.Field)(f)
-	return fmt.Sprintf("%v", fl.Value())
+	if f.IsZero() {
+		return ""
+	}
+
+	return fmt.Sprintf("%v", f.field.Value())
 }
 
 func (f *fieldValue) Get() interface{} {
-	fl := (*structs.Field)(f)
-	return fl.Value()
+	if f.IsZero() {
+		return nil
+	}
+
+	return f.field.Value()
+}
+
+func (f *fieldValue) IsZero() bool {
+	return f.field == nil
 }
 
 // This is an unexported interface, be careful about it.
 // https://code.google.com/p/go/source/browse/src/pkg/flag/flag.go?name=release#101
 func (f *fieldValue) IsBoolFlag() bool {
-	fl := (*structs.Field)(f)
-	if fl.Kind() == reflect.Bool {
-		return true
-	}
-	return false
+	return f.field.Kind() == reflect.Bool
 }
 
 // flagUsageDefault is the default "FlagUsageFunc" use in filling out

